src/connection/eagerlyConnect.ts

```diff
@@ -0,0 +1,36 @@
+import { Connector } from '@web3-react/types'
+
+import { getConnection, gnosisSafeConnection, networkConnection } from './index'
+import { ConnectionType, selectedWalletKey, toConnectionType } from './types'
+
+async function connect(connector: Connector, type: ConnectionType) {
+  performance.mark(`web3:connect:${type}:start`)
+  try {
+    if (connector.connectEagerly) {
+      await connector.connectEagerly()
+    } else {
+      await connector.activate()
+    }
+    return true
+  } catch (error) {
+    console.debug(`web3-react eager connection error: ${error}`)
+    return false
+  } finally {
+    performance.measure(`web3:connect:${type}`, `web3:connect:${type}:start`)
+  }
+}
+
+connect(gnosisSafeConnection.connector, ConnectionType.GNOSIS_SAFE)
+connect(networkConnection.connector, ConnectionType.NETWORK)
```

Comment:
is it ok to call these together? will networkConnection overwrite the gnosis connection?

Reply:
nvm, i see this is just copied from `useEagerlyConnect`. if it was working before then it should be good now.

Reply:
they'll both attempt to connect. if both succeed (which is possible), web3-react will use the one that comes first in the ordered list, which should be gnosis safe. as you said, this is code that's already in there.